koptional/src/main/kotlin/com/gojuno/koptional/Optional.kt

```diff
@@ -1,3 +1,5 @@
+@file:JvmName("Optional")
```

Comment:
I 100% agree with this change, the only problem is that it's a breaking change for Java consumers (class name change) and we'll need to make a 2.x release for that

However I'm not opposed to that

@dmitry-novikov @ming13 @nostra13 are you ok with 2.x release? 

If yes — we'll need separate issue and PRs for that (Group name will also need to be updated to `com.gojuno.koptional2`)

Reply:
I'm OK with bumping major version to indicate breaking change, but against package change because is will affect a lot of consumers and most of them are Kotlin based. So I would better force Java consumers to refactor their codebase because of class name change =)

Reply:
@artem-zinnatullin, you are too much in all this _package change for no reason_.

Reply:
Let’s make a minor release and that’s it. Koptional is a Kotlin-first library, I see no harm in changing Java _compatibility_ as a minor change. It is OK to change since it is no radical. Otherwise every library should follow Chrome or Firefox versions scheme and have versions in double or triple digits since API change all the time and hey, there are artifact versions exactly for that. Artem just has some kind of fashion taste in bumping package names (even for internal libraries with a single user, yep) :-)

Reply:
Given that Koptional is Kotlin-first, I'd agree that this is a minor change, even though it breaks Java consumers. Changing the package name will require Kotlin consumers to change imports everywhere, and bumping the major version feels excessive for such a small change - compare it with RxJava 2 vs RxJava 1 or Dagger 2 vs Dagger 1. Even though it's breaking, it's trivial to fix.

Reply:
It's trivial to fix for sure (basically find + replace for package name)

However because Koptional was initially designed to be very minimalistic it might be used by other libraries which means that in such a case we're breaking compatibility between user's code and other library. 

That's why i suggest both major version bump and groupid update :)

GitHub search results: https://github.com/search?utf8=%E2%9C%93&q=com.gojuno.koptional&type=Code

Reply:
Search shows no usage from Java code, so...

Reply:
Which doesn't mean there are no private Java libs that depend on class name…

This change is still breaking even for @Egorand himself because they use Koptional from Java (that's the point of PR), which means that we **should** at least **bump major version** https://semver.org/

Do we need to add number to `groupId` and package name? 
I think yes, it's very trivial to migrate to new one in same commit as dependency update and we allow users to migrate their own code and dependencies gradually in case of class-name problem

I might not agree with everything Jake thinks but http://jakewharton.com/java-interoperability-policy-for-major-version-updates/ is very good

Reply:
Let’s assume we have more Kotlin users than Java users. Changing a package name will force all users to do a Find + Replace. Changing a Java name will do the same for Java users only. I’m in for the version change, but against the package change. I totally understand all your points and, believe me, I would be on your side regarding anything related to these kind of changes. But! This PR does not introduce any behavior changes. If `.toNullable` was changed, that’s for sure forces the package change. At the same time, the library is so tiny I cannot even believe we are arguing about this 😆